bin/rad-issues

```diff
@@ -38,56 +38,109 @@
     (load! "rad/monadic/names.rad")
     (def names (ref (chain/load-chain! name-base)))
-    (def mk-key (fn [v]
-      (match v
-        ['key 'val] [(string-append (show key) " - " (lookup :title val)) val])))
-    (def iss (dict-from-seq (map mk-key (seq (list-issues chain-ref)))))
-    (def selected (fzf-select-by-key! iss))
-    (def comment-divider "---")
-    (def username
+    (def add-username
       (fn [d]
-        (catch
-          'any
-          (lookup :github-username d)
-          (fn [x]
-            (match (lookup-key names (lookup :author d))
-              :nothing "???"
-              'name    name)))))
-    (def issue-number (fn [d] (lookup :number d)))
-    (def body (fn [d] (lookup :body d)))
-    (def pretty-comment (fn [c]
-      (string-append
-        "### " (username c) " [" (lookup :created-at c) "]\n\n"
-        (body c) "\n")))
-    (def pretty-issue (fn [i]
-      (string-replace "\r" "" (string-append
-         "\n" comment-divider "\n"
-         "---Add comment above the line.\n\n"
-         "# Issue " (show (issue-number i)) ": " (lookup :title i) "\n\n"
-         "**State:** " (if (eq? (lookup :state i) :open) "open" "closed") "\n"
-         "**Labels:** " (show (lookup :labels i)) "\n\n"
-         (body i) "\n\n"
-         "Comments\n"
-         "--------\n\n"
-         (string/unlines (map pretty-comment (lookup :comments i)))
-    ))))
-    (def edited-issue (edit-in-editor! (pretty-issue selected)))
-    (def extract-comment
-      (fn [xs]
-        (string/unlines (take-while (fn [x] (not (eq? x comment-divider))) (string/lines xs)))))
-    (def send-comment
-      (fn [xs]
-        (def comment (extract-comment edited-issue))
-        (if (eq? comment "")
-          "Issue was not updated."
-          (do (simple-add-comment! chain-ref (issue-number selected) comment)
-            (string-append "Sent comment: " comment)))))
-    (put-str! (send-comment edited-issue))))
+        (def n (catch
+                   'any
+                   (lookup :github-username d)
+                 (fn [x]
+                   (match (lookup-key names (lookup :author d))
+                          :nothing "???"
+                          'name    name))))
+        (insert :username n d)))
+
+    (def add-username-issue
+      (fn [i]
+        (add-username
+         (insert :comments (map add-username (lookup :comments i)) i))))
+
+    (def mk-key
+      (fn [v]
+        (match v
+               (/as 'a {:activity :new-issue
+                        :issue    {:number 'n
+                                   :title 'tit}})
+               [(string-append "[new] Issue " (show n) " - " tit) a]
+               (/as 'a {:activity :new-comments
+                        :issue    {:number 'n
+                                   :title 'tit}})
+               [(string-append "[new comments] Issue " (show n) " - " tit) a])))
+
+    (def items (map mk-key (recent-activity chain-ref last-seen)))
```

Comment:
Shall we remove the word `"new"`?

Reply:
What I meant was that currently the command `bin/rad-issues list chain-name` returns all issues (as it should) but all of them are either marked as `new` or `new comment` (which I think should not happen). Regarding performance, if "only" the `list` command is given, I think it would be better, if the `recent-activity` call and extra filtering are omitted.

Reply:
I think performance is bad regardless and needs to be fixed in some other way (by caching the state of the chain).

But I see what you mean about the UI, will change.